core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -182,20 +183,23 @@ protected void dropPartition(int specId, StructLike partition) {
     // dropping the data in a partition also drops all deletes in the partition
     filterManager.dropPartition(specId, partition);
     deleteFilterManager.dropPartition(specId, partition);
+    changedPartitions.add(partition);
```

Comment:
The spec ID should matter, right?

Reply:
Yes, I think this spec ID should matter.